src/PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.cpp

```diff
@@ -0,0 +1,261 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/GrMhd/RiemannProblem.hpp"
+
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/LorentzFactor.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Math.hpp"  // IWYU pragma: keep
+
+namespace grmhd::AnalyticData {
+
+RiemannProblem::RiemannProblem(
+    const double adiabatic_index, const double left_rest_mass_density,
+    const double right_rest_mass_density, const double left_pressure,
+    const double right_pressure,
+    const std::array<double, 3>& left_spatial_velocity,
+    const std::array<double, 3>& right_spatial_velocity,
+    const std::array<double, 3>& left_magnetic_field,
+    const std::array<double, 3>& right_magnetic_field, const double lapse,
+    const double shift) noexcept
+    : equation_of_state_(adiabatic_index),
+      adiabatic_index_(adiabatic_index),
+      left_rest_mass_density_(left_rest_mass_density),
+      right_rest_mass_density_(right_rest_mass_density),
+      left_pressure_(left_pressure),
+      right_pressure_(right_pressure),
+      left_spatial_velocity_(left_spatial_velocity),
+      right_spatial_velocity_(right_spatial_velocity),
+      left_magnetic_field_(left_magnetic_field),
+      right_magnetic_field_(right_magnetic_field),
+      lapse_(lapse),
+      shift_(shift) {}
+
+RiemannProblem::RiemannProblem(CkMigrateMessage* /*unused*/) noexcept {}
+
+void RiemannProblem::pup(PUP::er& p) noexcept {
+  p | adiabatic_index_;
+  p | left_rest_mass_density_;
+  p | right_rest_mass_density_;
+  p | left_pressure_;
+  p | right_pressure_;
+  p | left_spatial_velocity_;
+  p | right_spatial_velocity_;
+  p | left_magnetic_field_;
+  p | right_magnetic_field_;
+  p | equation_of_state_;
+  p | background_spacetime_;
```

Comment:
[nitpick] move eos and background to top of list so the pup order matches declaration order + initialization order, makes it easier to read.